File: 백준/Bronze/27161. 크레이지 타임/크레이지 타임.py
import sys

input = sys.stdin.readline

N = int(input().strip())
time = 1
result = list()
isReverse = False
answer = "NO"

for _ in range(N):
    type, num = map(str, input().strip().split())
    if type == "HOURGLASS":
        if time != int(num):
            isReverse = False if isReverse else True
        answer = "NO"
    else:
        answer = "YES" if time == int(num) else "NO"
    
    result.append(f"{time} {answer}")

    if isReverse:
        time -= 1
    else:
        time += 1
    if time > 12:
        time -= 12
    elif time < 1:
        time += 12

sys.stdout.write("\n".join(result) + "\n")

File: 백준/Bronze/27161. 크레이지 타임/크레이지 타임.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine().trim());
        int time = 1;
        StringBuilder sb = new StringBuilder();
        boolean isReverse = false;
        String answer = "NO";

        for(int i = 0; i < N; i++){
            String[] line = br.readLine().split(" ");
            if(line[0].equals("HOURGLASS")){
                if(time != Integer.parseInt(line[1])){
                    isReverse = isReverse ? false : true;
                }
                answer = "NO";
            }else{
                answer = time == Integer.parseInt(line[1]) ? "YES" : "NO";
            }
            sb.append(time).append(" ").append(answer).append('\n');
            if (isReverse) time--;
            else time++;
            if (time > 12) time -= 12;
            else if (time < 1) time += 12;
        }
        br.close();

        System.out.print(sb.toString());
    }
}